dowhy/causal_refuters/linear_sensitivity_analysis.py

```diff
@@ -0,0 +1,435 @@
+
+            r2uwj_wt = self.frac_strength_treatment * (r2twj_w ** 2) / ((1 - self.frac_strength_treatment * r2twj_w) * (1 - r2twj_w))
+            if(any(val >= 1 for val in r2uwj_wt)):
+                raise ValueError("r2uwj_wt can not be >= 1. Try a lower frac_strength_treatment value")
+
+            self.r2yu_tw = ((np.sqrt(self.frac_strength_outcome) + np.sqrt(r2uwj_wt)) / np.sqrt(1 - r2uwj_wt)) ** 2 * (r2ywj_tw / (1 - r2ywj_tw))
+            if(any(val > 1 for val in self.r2yu_tw)):
+                for i in range(len(self.r2yu_tw)):
+                    if self.r2yu_tw[i]>1:
+                        self.r2yu_tw[i]=1
+                self.logger.warning("Warning: r2yu_tw can not be > 1. Try a lower frac_strength_treatment. Setting r2yu_tw to 1")
+
+            #Compute bias adjusted terms
+        
+        self.benchmarking_results = self.compute_bias_adjusted(self.r2tu_w, self.r2yu_tw)
+        self.bias_adjusted_estimate = self.benchmarking_results['bias_adjusted_estimate']
+        self.bias_adjusted_se = self.benchmarking_results['bias_adjusted_se']
+        self.bias_adjusted_t = self.benchmarking_results['bias_adjusted_t']
+        self.bias_adjusted_lower_CI = self.benchmarking_results['bias_adjusted_lower_CI']
+        self.bias_adjusted_upper_CI = self.benchmarking_results['bias_adjusted_upper_CI']
+        
+        self.plot()
+
+        self.stats = {
+            'estimate' : self.estimate,
+            'standard_error' : self.standard_error,
+            'degree of freedom': self.degree_of_freedom,
+            't_statistic' : self.t_stats,
+            'r2yt_w' : partial_r2,
+            'partial_f2' : self.partial_f2,
+            'robustness_value' : RVq,
+            'robustness_value_alpha' : RVqa 
+        }
+
+        self.benchmarking_results = pd.DataFrame.from_dict(self.benchmarking_results)
+        return self
+
+    def plot_estimate(self, r2tu_w, r2yu_tw):
+        """
+        Computes the contours, threshold line and bounds for plotting estimates.
+        Contour lines (z - axis) correspond to the adjusted estimate values for different values of r2tu_w (x) and r2yu_tw (y).
+        :param r2tu_w: hypothetical partial R^2 of confounder with treatment(x - axis)
+        :param r2yu_tw: hypothetical partial R^2 of confounder with outcome(y - axis)
+
+        :returns:
+        contour_values : values of contour lines for the plot
+        critical_estimate : threshold point 
+        estimate_bounds : estimate values for unobserved confounders (bias adjusted estimates)
+        """
+
+        critical_estimate = self.h0
+        contour_values = np.zeros((len(r2yu_tw), len(r2tu_w)))
+        for i in range(len(r2yu_tw)):
+            y = r2tu_w[i]
+            for j in range(len(r2tu_w)):
+                x = r2yu_tw[j]
+                benchmarking_results = self.compute_bias_adjusted(r2tu_w = x, r2yu_tw = y)

```

Comment:
isn't the benchmarking_results already computed in `perform_analysis`? Can you reuse them here?

Reply:
It is computed for the frac strength of treatment with confounders based on benchmark covariates.
These are the hypothetical r^2 values we use to generate the plots. we need to recalculate the estimates for them.